### ISSUE TITLE
`no-unused-props` reports false positive

### ISSUE THREAD

**Issue by baseballyama** [sveltejs/eslint-plugin-svelte#1159]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.23.0

### What version of `eslint-plugin-svelte` are you using?

3.3.3

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```svelte
<script lang="ts">
	type Props = {
		foo: string;
	};

	const props: Props = $props();
	console.log(props);
</script>

```

### What did you expect to happen?

No lint error

### What actually happened?

![Image](https://github.com/user-attachments/assets/def15669-6341-47f2-804a-abe8b9a5ec0a)

### Link to **GitHub Repo** with Minimal Reproducible Example

```svelte
<script lang="ts">
  interface Props {
    foo: string;
  }
  const props: Props = $props();

  console.warn(props);
</script>
```
